File: seed/csharp-model/trace/src/SeedTraceClient/V2/V3/Problem/TestCaseFunction.cs
using SeedTraceClient.V2.V3;
using System.Text.Json.Serialization;

namespace SeedTraceClient.V2.V3;

public class TestCaseFunction
{
    public class _WithActualResult : TestCaseWithActualResultImplementation
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "withActualResult";
    }
    public class _Custom : VoidFunctionDefinition
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "custom";
    }
}

File: seed/rust-sdk/trace/src/types/test_case_function.rs
use crate::test_case_with_actual_result_implementation::TestCaseWithActualResultImplementation;
use crate::void_function_definition::VoidFunctionDefinition;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(tag = "type")]
pub enum TestCaseFunction {
        WithActualResult {
            #[serde(flatten)]
            data: TestCaseWithActualResultImplementation,
        },

        Custom {
            #[serde(flatten)]
            data: VoidFunctionDefinition,
        },
}
